src/methods.rs

```diff
@@ -0,0 +1,87 @@
+use crate::RequestBuilder;
+
+/// Trait that adds http request methods.
```

Comment:
I think this could use better documentation about what exactly this is. I think that is my primary concern at this point, it may be confusing for someone to see a trait named "Methods" in the docs without more exposition.

Reply:
Agreed, but not sure what to write. Suggestions appreciated